pkg/splunk/enterprise/util.go

```diff
@@ -2131,7 +2130,9 @@ func updateCRStatus(ctx context.Context, client splcommon.ControllerClient, orig
 }
 
 // fetchCurrentCRWithStatusUpdate returns a CR (fresh Read) with latest status copied
-func fetchCurrentCRWithStatusUpdate(ctx context.Context, client splcommon.ControllerClient, origCR splcommon.MetaObject) (splcommon.MetaObject, error) {
+// Use this API to update the CR status message with an error if any. This aviods multiple
+// hops of CR specific logic to determine CR type.
+func fetchCurrentCRWithStatusUpdate(ctx context.Context, client splcommon.ControllerClient, origCR splcommon.MetaObject, crError *error) (splcommon.MetaObject, error) {
```

Comment:
I do not think you need error pointer receiver to be passed here , you are not manipulating the error message. its just for read only.

Reply:
standard golang error is just an interface that takes string error message

Reply:
Per this [comment](https://github.com/splunk/splunk-operator/pull/1335#pullrequestreview-2113664379) closing this.